apps/cowswap-frontend/src/modules/limitOrders/services/tradeFlow/index.tsx

```diff
@@ -85,18 +86,23 @@ export async function tradeFlow(
       signer: provider.getSigner(),
       validTo,
     })
+
     logTradeFlow('LIMIT ORDER FLOW', 'STEP 5: add pending order step')
-    addPendingOrderStep(
-      {
-        id: orderId,
-        chainId: chainId,
-        order: {
-          ...order,
-          isHidden: !allowsOffchainSigning,
-        },
-      },
-      dispatch
-    )
+
+    if (allowsOffchainSigning) {
+      addSnackbar({
+        id: 'pending-limit-order',
+        icon: 'success',
+        content: (
+          <PendingOrderNotification
+            orderId={orderId}
+            kind={postOrderParams.kind}
+            inputAmount={postOrderParams.inputAmount}
+            outputAmount={postOrderParams.outputAmount}
+          />
+        ),
+      })
```

Comment:
How does it handle the hidden part?

Keep in mind that adding the pending order to the store here is not only for the notification.
At this step there is also info not available from the API.

Reply:
I've refactored that part. Decided not to move in this direction because it cause too much refactoring